sdk/python/feast/infra/offline_stores/offline_store.py

```diff
@@ -179,7 +179,12 @@ def on_demand_feature_views(self) -> List[OnDemandFeatureView]:
         pass
 
     @abstractmethod
-    def persist(self, storage: SavedDatasetStorage, allow_overwrite: bool = False):
+    def persist(
+        self,
+        storage: SavedDatasetStorage,
+        allow_overwrite: Optional[bool] = False,
+        timeout: Optional[int] = 1800,
```

Comment:
I'd prefer the default value to be `None` - implementations that support a timeout can handle it explicitly